### PR TITLE
Handle ProductMedia delete in task

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because we want to handle product media deletion in task to prevent 
long ProductMediaDelete mutation executions.
Deleting ProductMedia will switch new BooleanField `to_remove` into `True` and run a task to clear out all connected media files from s3 and delete ProductMedia object from db.
Deleting Product will send pre_delete signal that will switch flag and run the task. (pre delete will be triggered as well when Product will be deleted by CASCADE)

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4783 Long time to execute queries in ProductMediaDelete](https://app.clickup.com/t/2549495/SALEOR-4783)

File: saleor/graphql/product/mutations/products.py [hunk 1]
```diff
@@ -1700,8 +1701,19 @@ class Meta:
...
+                        "Media not found.",
+                        code=ProductErrorCode.NOT_FOUND,
+                    )
+                }
+            )
         media_id = media_obj.id
-        media_obj.delete()
+        media_obj.to_remove = True
+        media_obj.save(update_fields=["to_remove"])
+        delete_product_media_task.delay(media_id)
```
Comment (L3str4nge) [inline saleor/graphql/product/mutations/products.py:1716]: I would run the task only if `ProductMedia` exists. Basically move line this line from task here `product_media = ProductMedia.objects.filter(pk=media_id, to_remove=True).first()`

Comment (IKarbowiak) [reply saleor/graphql/product/mutations/products.py:1716]: AFAIK we cannot pass media object to the celery task.

**Review by IKarbowiak — APPROVED** [review]

**Review by Qasem-h — APPROVED** [review]

File: saleor/graphql/product/mutations/products.py (hunk 1 above)
Comment (jakubkuc) [reply saleor/graphql/product/mutations/products.py:1716]: Yes, just like @IKarbowiak mentioned. That is why I passed media id media can be fetched by task itself

**Review by maarcingebala — APPROVED** [review]

File: saleor/core/tasks.py [hunk 2]
```diff
@@ -21,3 +24,16 @@ def delete_event_payloads_task():
     EventPayload.objects.filter(
         deliveries__isnull=True, created_at__lte=event_payload_delete_period
     ).delete()
+
+
+@app.task(
+    autoretry_for=(ClientError,),
```
Comment (fowczarek) [inline saleor/core/tasks.py:30]: Should we auto-retry only on `ClientError`?
@korycins

**Review by fowczarek — APPROVED** [review]

File: saleor/core/tasks.py (hunk 2 above)
Comment (korycins) [reply saleor/core/tasks.py:30]: Do we support only S3 for storing images? It looks like we took an AWS error from the lower layer used in versatile.

**Pull request merged**